service/dap/server_test.go

```diff
@@ -5353,11 +5353,37 @@ func TestNoDebug_AcceptNoRequestsButDisconnect(t *testing.T) {
 
 		// Disconnect request is ok
 		client.DisconnectRequestWithKillOption(true)
-		client.ExpectOutputEventTerminating(t)
-		client.ExpectOutputEventRegex(t, fmt.Sprintf(daptest.ProcessExited, "(-1|1)"))
-		client.ExpectTerminatedEvent(t)
```

Comment:
The way the code is now written, it's really only this one TerminatedEvent that can happen before or after the DisconnectRequest. Everything else should happen in the right order, so I think it is safe to leave the OutputEvent checks as-is outside of the new loop. Have you actually seen this flake in different ways?

Reply:
I would prefer that DAP test relied less on checking a precise order for events/responses, when it doesn't matter. In this test for example, it doesn't matter when the output events happen or where the terminated event happens before or after the response to the disconnect request: all orderings are valid from the point of view of DAP.
The test, as it is, is detecting implementation changes that are inconsequential to clients.